src/picologging/logger.cxx

```diff
@@ -22,6 +22,39 @@ int getEffectiveLevel(Logger*self){
     return LOG_LEVEL_NOTSET;
 }
 
+void setEnabledBasedOnEffectiveLevel(Logger* logger) {
+    logger->enabledForDebug = false;
+    logger->enabledForInfo = false;
+    logger->enabledForWarning = false;
+    logger->enabledForError = false;
+    logger->enabledForCritical = false;
+    switch (logger->effective_level){
+        case LOG_LEVEL_DEBUG:
+            logger->enabledForDebug = true;
+        case LOG_LEVEL_INFO:
+            logger->enabledForInfo = true;
+        case LOG_LEVEL_WARNING:
+            logger->enabledForWarning = true;
+        case LOG_LEVEL_ERROR:
+            logger->enabledForError = true;
+        case LOG_LEVEL_CRITICAL:
+            logger->enabledForCritical = true;
+    }
+}
+
+void setEffectiveLevelOfChildren(Logger* logger, unsigned short level) {
+    for (int i = 0; i < PyList_GET_SIZE(logger->children); i++) {
+        PyObject *child_logger = PyList_GET_ITEM(logger->children, i); // borrowed ref
+        if (((Logger*)child_logger)->level == LOG_LEVEL_NOTSET) {
```

Comment:
Should really add a type check here (`Logger_Check(child_logger)`) before accessing it's properties, unless we're 100% sure that `logger->children` cannot contain an item which is not a picologging `Logger` instance?

Reply:
I thought about that and I don't think it's possible, since the Append to children happens inside Logger_set_parent:

    if (PySequence_Contains(((Logger*)self->parent)->children, (PyObject*)self) == 0){
        PyList_Append(((Logger*)self->parent)->children, (PyObject*)self);
    }

So that wouldn't be callable on something that wasn't a logger. I'm happy to add a check if that's best practice though, doesn't hurt, just more code not covered by test coverage.

Reply:
That's fine then